### PR TITLE
ZTS: Move more tests out of common.run

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

These tests won't run on all platforms as currently implemented:
 * add_nested_replacing_spare (needs zed)
 * fault (needs zed)
 * mmp (needs multihost_history)
 * umount_unlinked_drain (needs procfs)
 * zpool_expand (needs udev events and zed)
 * zpool_reopen (needs scsi_debug)
 * zvol_swap_003_pos (needs to modify vfstab)
 * zvol_swap_00[56]_pos (needs swaphigh/swaplen)

### Description

Move the zvol_swap_* tests to sunos.run, the rest to linux.run.

### How Has This Been Tested?

ZTS on FreeBSD.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
> This seems reasonable to avoid known failures on FreeBSD due to missing bits. Is the slightly longer term plan for FreeBSD to eventually adapt most of these tests so they can run on the platform?

Yes. In particular:
* The zpool_reopen tests seem like they will be doable using gnop on FreeBSD instead of scsi_debug, but it may take some extensive restructuring in those tests for that to work.
* The mmp tests make use of things in the kernel which are just stubbed out on FreeBSD at the moment. I don't think the tests themselves will need too much beyond that, a few little things like moving /etc/hostid out of the way temporarily for the tests.

Right now zed is a question mark. We may adopt it or we may find a way to use FreeBSD's zfsd and devd and adapt the tests, I'm not sure which way it will go yet. For now our ZFS events are wired into devd rather than zpool events.

File: tests/runfiles/sunos.run [hunk 1]
```diff
@@ -37,3 +37,7 @@ tags = ['functional', 'cli_root', 'zpool_create']
 tests = ['zvol_misc_001_neg', 'zvol_misc_003_neg', 'zvol_misc_004_pos',
     'zvol_misc_005_neg', 'zvol_misc_006_pos']
 tags = ['functional', 'zvol', 'zvol_misc']
+
+[tests/functional/zvol/zvol_swap:illumos]
+tests = ['zvol_swap_003_pos', 'zvol_swap_005_pos', 'zvol_swap_006_pos']
+tags = ['functional', 'zvol', 'zvol_swap']
```
Comment (behlendorf) [inline tests/runfiles/sunos.run]: Making this illumos specific means we can remove them from the known skip list in `tests/test-runner/bin/zts-report.py`, and drop the `if is_linux || is_freebsd` check at the top of each test case.

**Comment by freqlabs** [comment]
Turned out I already had those changes, I just missed them when cherry-picking :)

**Comment by behlendorf** [comment]
@freqlabs this will need to be rebase now that #9872 has been merged.

**Pull request merged**